src/parser.cpp

```diff
@@ -277,7 +278,6 @@ namespace ada::parser {
 
             // If c is U+003F (?), then set url’s query to the empty string, and state to query state.
             if ((input_position != input_size) && (url_data[input_position] == '?')) {
-              url.query = "";
```

Comment:
This is fine, but in general, url.query.clear() is effectively free. It is true that url.query = "" is expensive.

Reply:
`clear()` is just like a couple of instructions... very, very fast.

Reply:
This was unnecessary because we are already assigning a new value in the query state